src/components/CardGrid.js

```diff
@@ -29,8 +29,18 @@ const CardGrid = ({ routerLocation, saveItem, resource }) => {
 };
 
 function mapStateToProps(state, ownProps) {
+    let res = [];
+    //Not the most efficient logic, but it works. Will have to optimize this later 
+    for (var i = 0, len1 = state.searchedResource.length; i < len1; i++) {
```

Comment:
this is fine for now- I think the word you might be looking for is an 'intersection' between 2 arrays.